pennylane/labs/tests/transforms/test_select_pauli_rot_phase_gradient.py

```diff
@@ -0,0 +1,169 @@
+    angle_wires = qml.wires.Wires([f"aux_{i}" for i in range(p)])
+    phase_grad_wires = qml.wires.Wires([f"qft_{i}" for i in range(p)])
+    work_wires = qml.wires.Wires([f"work_{i}" for i in range(p - 1)])
+
+    ops = _select_pauli_rot_phase_gradient(
+        phis,
+        control_wires=control_wires,
+        target_wire=wire,
+        angle_wires=angle_wires,
+        phase_grad_wires=phase_grad_wires,
+        work_wires=work_wires,
+    )
+
+    for op in ops:
+        assert op.name in ["QROM", "SemiAdder", "MultiControlledX", "Adjoint(QROM)"]
+
+
+def test_wire_validation():
+    """Test that an error is raised when phg wires are fewer than angle wires"""
+
+    circ = qml.tape.QuantumScript([qml.SelectPauliRot([0.5, 0.5], [0], 1)])
+
+    angle_wires = qml.wires.Wires([f"angle_{i}" for i in range(3)])
+    phase_grad_wires = qml.wires.Wires([f"phg_{i}" for i in range(2)])
+    work_wires = qml.wires.Wires([f"work_{i}" for i in range(2)])
+
+    with pytest.raises(
+        ValueError, match="phase_grad_wires needs to be at least as large as angle_wires"
+    ):
+        _ = select_pauli_rot_phase_gradient(
+            circ,
+            angle_wires=angle_wires,
+            phase_grad_wires=phase_grad_wires,
+            work_wires=work_wires,
+        )
+
+
+@pytest.mark.parametrize(
+    "rot_axis",
+    [
+        "Z",
+        "X",
+        "Y",
+    ],
+)
+def test_correctness_select_pauli_rot_phase_gradient(rot_axis):
+    """Test that decomposition produce the correct solution"""
+
+    precision = 4
+    wire = "targ"
+    angle_wires = [f"ang_{i}" for i in range(precision)]
+    phase_grad_wires = [f"phg_{i}" for i in range(precision)]
+    work_wires = [f"work_{i}" for i in range(precision - 1)]
+
+    def phase_gradient(wires):
+        # prepare phase gradient state
+        for i, w in enumerate(wires):
+            qml.H(w)
+            qml.PhaseShift(-np.pi / 2**i, w)
+
+    @partial(
+        select_pauli_rot_phase_gradient,
+        angle_wires=angle_wires,
+        phase_grad_wires=phase_grad_wires,
+        work_wires=work_wires,
+    )
+    @qml.qnode(qml.device("default.qubit"))
+    def select_pauli_rot_circ(phis, control_wires, target_wire):
+        phase_gradient(phase_grad_wires)  # prepare phase gradient state
+
+        for wire in control_wires:
+            qml.Hadamard(wire)
+
+        qml.SelectPauliRot(phis, control_wires, target_wire, rot_axis=rot_axis)
+
+        if rot_axis == "Y":
+            qml.adjoint(qml.S)(target_wire)
+
+        if rot_axis in ["X", "Y"]:
+            qml.Hadamard(target_wire)
+
+        qml.Select([qml.RZ(-phi, target_wire) for phi in phis], control=control_wires)
+
+        if rot_axis in ["X", "Y"]:
+            qml.Hadamard(target_wire)
+
+        if rot_axis == "Y":
+            qml.S(target_wire)
+
+        for wire in control_wires:
+            qml.Hadamard(wire)
+
+        return qml.probs(target_wire)
```

Comment:
Could use all probs to catch accidental transformations on the multiplexing wires as well?